examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -8,126 +8,403 @@
+
+        hand_positions = []
+        num_hands = 2
+        for i in range(num_hands):
+            hand_pos, _ = remote_gui_input.get_hand_pose(i)
+            if hand_pos:
+                hand_positions.append(hand_pos)
+        if len(hand_positions) == 0:
+            return
+        assert len(hand_positions) == num_hands
+
+        grasp_threshold = 0.2
+        grasped_objects_idxs = get_grasped_objects_idxs(self.get_sim())
+
+        remote_button_input = remote_gui_input.get_gui_input()
+
+        found_obj_idx = None
+        found_hand_idx = None
+
+        for i in range(len(self._target_obj_ids)):
+            # object is already grasped by Spot
+            if i in grasped_objects_idxs:
+                continue
+
+            this_target_pos = self._get_target_object_position(i)
+
+            for hand_idx in range(num_hands):
+                hand_pos = hand_positions[hand_idx]
+                if (this_target_pos - hand_pos).length() < grasp_threshold:
+                    # color = mn.Color3(0, 1, 0)  # green
+                    # box_half_size = 0.20
+                    # self._draw_box_in_pos(
+                    #     this_target_pos, color=color, box_half_size=box_half_size
+                    # )
+                    # show grasp
+                    self._add_target_object_highlight_ring(
+                        i, COLOR_GRASP_PREVIEW, radius=RADIUS_GRASP_PREVIEW
+                    )
+
+                    for key in self.get_grasp_keys_by_hand(hand_idx):
+                        if remote_button_input.get_key_down(key):
+                            found_obj_idx = i
+                            found_hand_idx = hand_idx
+                            break
+
+            if found_obj_idx is not None:
+                break
+
+        if found_obj_idx is None:
+            return
+
+        self._held_target_obj_idx = found_obj_idx
+        self._held_hand_idx = found_hand_idx
+        self.state_machine_agent_ctrl.cancel_fetch()
+        rom_obj = self._get_target_rigid_object(self._held_target_obj_idx)
+        rom_obj.motion_type = MotionType.KINEMATIC
+
+    def _update_held_and_try_throw_remote(self):
+        assert self._held_target_obj_idx is not None
+        assert self._held_hand_idx is not None
+
+        remote_gui_input = self._sandbox_service.remote_gui_input
+        remote_button_input = remote_gui_input.get_gui_input()
+
+        do_throw = False
+        for key in self.get_grasp_keys_by_hand(self._held_hand_idx):
+            if remote_button_input.get_key_up(key):
+                do_throw = True
+
+        rom_obj = self._get_target_rigid_object(self._held_target_obj_idx)
+
+        if do_throw:
+            rom_obj.motion_type = MotionType.DYNAMIC
+            rom_obj.collidable = True
+
+            hand_idx = self._held_hand_idx
+            history_len = remote_gui_input.get_history_length()
+            assert history_len >= 2
+            history_offset = 1
```

Comment:
The code below is using the recent history of controller poses to derive a throw velocity. Note `history_offset = 1` which means we ignore the very latest controller pose. In practice, I think we see that folks release the button almost exactly at the end of their throwing motion (or even after they've ended their throwing motion), so I find it's more reliable to ignore the last frame and thus derive throw velocity from only the beginning/middle of their throwing motion.